src/main/cpp/blaze_util.cc

```diff
@@ -74,6 +74,36 @@ bool GetNullaryOption(const char *arg, const char *key) {
   return true;
 }
 
+std::vector<const char*> SearchNaryOption(const vector<string>& args,
+                              const char *key) {
+  vector<const char *> values;
+  if (args.empty()) {
+    return values;
+  }
+
+  for (vector<string>::size_type i = 0; i < args.size(); ++i) {
+    if (args[i] == "--") {
+      // If the current argument is "--", all following args are target names.
+      // If 'key' was not found, 'value' is nullptr and we can return that.
+      // If 'key' was found exactly once, then 'value' has the value and again
+      // we can return that.
+      // If 'key' was found more than once then we could not have reached this
+      // line, because we would have broken out of the loop when 'key' was found
+      // the second time.
+      return values;
+    }
+    const char* result = GetUnaryOption(args[i].c_str(),
+                                        args[i + 1].c_str(),
```

Comment:
I could be wrong but following check
```
i < args.size()
```
does not protect from out of bound access via i + 1

Reply:
I agree it's a bit sketch, but it has something to do how `GetUnaryOption` is implemented: https://github.com/bazelbuild/bazel/blob/17e8b19297905abdc0c6ba715ef472e1f54d0b2b/src/main/cpp/blaze_util.cc#L47-L60 and I was hoping to minimize the surface area touched

Reply:
I can, however modify it to something like:
```
GetUnaryOption(args[i].c_str(),
                                        args[min(args.size() - 1) , i + 1].c_str(),
 ```

Reply:
Adjusted.